src/tabs.ts

```diff
@@ -0,0 +1,258 @@
+                this.DEFAULT_VALUES.column_size;
+    private _fashion_style:Array<ResponsiveVal> =
+                this.DEFAULT_VALUES.fashion_style;
+
+    constructor() {
+        super();
+    }
+    override attributeChangedCallback(
+                name:string, old_val:string, new_val:string) {
+        super.attributeChangedCallback(name, old_val, new_val);
+        if(name == 'animation') {
+            if(animate_defs[new_val])
+                this._animation_styles = {open: new_val, close: new_val}
+            else {
+                this._animation_styles = parseAttrVal(new_val);
+                this._animation_styles = Object.assign({}, {
+                    open:this._animation_styles?.show,
+                    close:this._animation_styles?.hide});
+            }
+        } else if(name == 'fashion') {
+            this._fashion_style = getResponsiveValues(
+                new_val, this.DEFAULT_VALUES.fashion);
+            let val = this._checkFashion();
+            this._changeFashionStyle(val);
+        } else if(name == "colsize")
+            this._column_size = parseAttrVal(new_val);
+    }
+    override render() {
+        if(this._internal_fashion == 'accordion') {
+            return html`
+<div id='accordion' @click=${this._titleClicked}>
+    <slot name='acc'></slot>
+</div>
+            `
+        } else {
+            let fashionColumn:any = {
+                display: 'grid',
+                'grid-template-columns': `
+                    ${this._column_size.title} ${this._column_size.body}`
+            }
+            return html`
+${this._animateStyles()}
+<div id="parent" class="parentDiv"
+    style=${this._internal_fashion=='column'?
+            styleMap(fashionColumn):''}>
+    <div id="title-tab" style=
+        "display:flex;
+         flex-direction:${this._internal_fashion=='column'?
+            'column':'row'}">
+        <slot name="tab-title" @click=${this._titleClicked}>
+        </slot>
+    </div>
+    <div id="body-tab">
+        <slot id="body-slot" name="tab-body"></slot>
+    </div>
+</div>
+            `
+        }
+    }
+    override connectedCallback() {
+        super.connectedCallback();
+        this.renderRoot.addEventListener("slotchange", (e) => {
+            this._slotChangedEvent(e)});
+        window.addEventListener("resize", this._resizeListener);
+    }
+    override disconnectedCallback() {
+        window.removeEventListener("resize", this._resizeListener);
+    }
+    override firstUpdated() {
+        if(!this.animation)
+            this.animationDur = 0;
+        this._openActive();
+    }
+    private _slotChangedEvent(event:Event) {
+        if(!event.target)
+            return;
+        let nodes = (event.target as HTMLSlotElement).assignedElements();
+        let name = (event.target as HTMLSlotElement)?.name;
+        if(name == 'tab-body') {
+            for(let i of nodes)
+                if(!i.classList.contains(this.activecls))
```

Comment:
Imagine a case where the tabs get added dynamically, then we have two cases
1. If the newly added tab-title and tab-body carries the active cls, then it should be hidden if an already active tab is present - when the current fashion is row/column
2. If the current fashion is accordion, then we need to add a new accordion item.